docker/Dockerfile

```
@@ -58,39 +68,20 @@ RUN if [ "$(uname -m)" != "aarch64" ]; then \
     && \
     rm -rf /var/lib/apt/lists/* /tmp/*
 
-# TODO: just keep the backtrace, colorbt, rust_log, and cargo_home variables as those are the only needed at build time.
-# Build arguments and variables set to change how tests are run, tracelog levels,
-# and Network to be used (Mainnet or Testnet)
+# Build arguments and variables set for tracelog levels and debug information
 #
 # We set defaults to all variables.
 ARG RUST_BACKTRACE
 ENV RUST_BACKTRACE=${RUST_BACKTRACE:-0}
```

I'd like to turn on backtraces, so we can to diagnose bugs in Zebra tests. This requires some environmental variables to be set to `1`.

For example, I can't make sure I've fixed the panic bugs in #7170 without a backtrace, because this is all the output I get:
> Message:  panic in ReadRequest::Tip: JoinError::Cancelled(Id(23437))
> Location: zebra-state/src/service.rs:1208
> Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.

https://github.com/ZcashFoundation/zebra/actions/runs/5470776835/jobs/9961459769#step:5:347

```suggestion
ENV RUST_BACKTRACE=${RUST_BACKTRACE:-1}

ARG RUST_LIB_BACKTRACE
ENV RUST_LIB_BACKTRACE=${RUST_LIB_BACKTRACE:-1}
```

---

If this is meant to be set at runtime (tests and/or production), we should handle those in the pipelines with `vars.RUST_BACKTRACE` and the others, as we've done here: https://github.com/search?q=repo%3AZcashFoundation%2Fzebra%20vars.RUST_BACKTRACE&type=code

---

That works for our environment, but we also want these variables set to get better bug reports from users.

Do our CI environment variables get propagated into the production Docker images as defaults?

---

Our CI environment variables _can_ get propagated to the **deployments** if those are set in the workflow (globally or on the specific job/task) and added, for example, here:
- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/continous-delivery.yml#L279

Or here, using `-e`:
- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/deploy-gcp-tests.yml#L256

If we want defaults (at runtime) for our production images, we would add them in the entrypoint as we're doing here:
- https://github.com/ZcashFoundation/zebra/blob/main/docker/runtime-entrypoint.sh#L20

If we want defaults (at build time) for our production and/or tests images, we should add them in Dockerfile as `ARG` and `ENV` here:
- https://github.com/ZcashFoundation/zebra/blob/main/docker/Dockerfile#L66
- https://github.com/ZcashFoundation/zebra/blob/main/.github/workflows/build-docker-image.yml#L159

**Summary**
- Setting variables–_without_ defaults–at `runtime` is the easiest, those doesn't have to be defined in the `Dockerfile` nor the `entrypoint`
- Setting _default_ variables at `runtime` just requires to set them in the `entrypoint`
- Setting _default_ variables at build time requires them to be `ARG` (`--build-arg`) and `ENV`, as that's the only way to set them.

**Note**: `NETWORK` is not required as a variable at build time, so we can easily remove those `ARG`, they still exist as our pipeline was built expecting this variable as an `ARG` (that was a wrong understanding on my side as when this variable was actually required).

---

Thanks for the explanation! What we need to debug panics and other bugs is for the 3 backtrace environmental variables to be set in ZF tests, ZF deployments, and user-run images.

Since they only change the output during panics, it's unlikely that users will want to turn them off, so making them user-configurable is optional.

I'm focused on other work right now, so I'm going to leave the details to you both.

---

As making this change will require further changes in our pipeline (aside from the Dockerfile), I'll make a different PR for that